packages/calcite-components/src/components/date-picker-month-header/date-picker-month-header.tsx

```diff
@@ -124,66 +163,125 @@ export class DatePickerMonthHeader {
+            [CSS.monthYearContainer]: true,
+            [CSS.monthYearContainerReverse]: reverse,
+            [CSS.rangeCalendar]: !!this.position,
+          }}
         >
-          <calcite-icon flip-rtl icon={ICON.chevronLeft} scale={getIconScale(this.scale)} />
-        </a>
-        <div class={{ text: true, [CSS.textReverse]: reverse }}>
-          <Heading class={CSS.month} level={this.headingLevel}>
-            {localizedMonth}
-          </Heading>
-          <span class={CSS.yearWrap}>
-            <input
-              aria-label={messages.year}
-              class={{
-                year: true,
-                [CSS.yearSuffix]: !!suffix,
-              }}
-              inputmode="numeric"
-              maxlength="4"
-              minlength="1"
-              onChange={this.onYearChange}
-              onInput={this.onYearInput}
-              onKeyDown={this.onYearKey}
-              pattern="\d*"
-              ref={(el) => (this.yearInput = el)}
-              type="text"
-              value={localizedYear}
-            />
-            {suffix && <span class={CSS.suffix}>{suffix}</span>}
-          </span>
+          {this.renderMonthPicker(months, activeMonth)}
+          {this.renderYearPicker()}
+        </div>
+        {!this.position && (
+          <div class={{ [CSS.chevronContainer]: true }}>{this.renderChevron("left")}</div>
+        )}
+        <div class={{ [CSS.chevronContainer]: true }}>
+          {this.position !== "start" && this.renderChevron("right")}
         </div>
-        <a
-          aria-disabled={`${this.nextMonthDate.getMonth() === activeMonth}`}
-          aria-label={messages.nextMonth}
-          class={CSS.chevron}
-          href="#"
-          onClick={this.nextMonthClick}
-          onKeyDown={this.nextMonthKeydown}
-          role="button"
-          tabindex={this.nextMonthDate.getMonth() === activeMonth ? -1 : 0}
-        >
-          <calcite-icon flip-rtl icon={ICON.chevronRight} scale={getIconScale(this.scale)} />
-        </a>
       </Fragment>
     );
   }
 
+  private renderMonthPicker(months: DateLocaleData["months"], activeMonth: number): VNode {
+    const monthData = this.monthAbbreviations ? months.abbreviated : months.wide;
+    return (
+      <calcite-select
+        class={CSS.monthPicker}
+        label="Month menu"
+        onCalciteSelectChange={this.handleMonthChange}
+        ref={(el) => (this.monthPickerEl = el)}
+        width="auto"
+      >
+        {monthData.map((month: string, index: number) => {
+          return (
+            <calcite-option
+              disabled={this.isMonthInRange(index)}
+              selected={index === activeMonth}
+              value={month}
+            >
+              {month}
+            </calcite-option>
+          );
+        })}
+      </calcite-select>
+    );
+  }
+
+  private renderYearPicker(): VNode {
+    return (
+      <calcite-select
+        class={CSS.yearPicker}
+        label="Year menu"
```

Comment:
This should use `messages`. Applies to other labels.

Reply:
Good Catch! This needs to be updated to use messages as translations are back.